schematic/manifest/generator.py

```diff
@@ -767,19 +766,54 @@ def set_dataframe_by_url(
         Returns:
             ps.Spreadsheet: A Google Sheet object.
         """
+
         # authorize pygsheets to read from the given URL
         gc = ps.authorize(custom_credentials=self.creds)
 
         # open google sheets and extract first sheet
         sh = gc.open_by_url(manifest_url)
         wb = sh[0]
 
+        # Handle scenario when existing manifest does not match new
+        #       manifest template due to changes in the data model:
+        #
+        # the sheet column header reflect the latest schema
+        # the existing manifest column-set may be outdated
+        # ensure that, if missing, attributes from the latest schema are added to the
+        # column-set of the existing manifest so that the user can modify their data if needed 
+        # to comply with the latest schema
+
+        # get headers from existing manifest and sheet
+        wb_header = wb.get_row(1)
+        manifest_df_header = manifest_df.columns
+        
+        # find missing columns in existing manifest
+        new_columns = set(wb_header) - set(manifest_df_header)
+
+        # find missing columns present in existing manifest but missing in latest schema
+        out_of_schema_columns = set(manifest_df_header) - set(wb_header)
+
+        # update existing manifest w/ missing columns, if any
+        if new_columns:
+            manifest_df = manifest_df.assign(**dict(zip(new_columns, len(new_columns)*[""])))
+
+        # sort columns in the updated manifest:
+        # match latest schema order
+        # move obsolete columns at the end
+        manifest_df = manifest_df[self.sort_manifest_fields(manifest_df.columns)]
+        manifest_df = manifest_df[[c for c in manifest_df if c not in out_of_schema_columns] + list(out_of_schema_columns)]
+        
         # The following line sets `valueInputOption = "RAW"` in pygsheets
         sh.default_parse = False
 
         # update spreadsheet with given manifest starting at top-left cell
         wb.set_dataframe(manifest_df, (1, 1))
 
+        # update validation rules (i.e. no validation rules) for out of schema columns
+        start_col = self._column_to_letter(len(wb_header)) # find start of out of schema columns
+        end_col = self._column_to_letter(len(manifest_df.columns) - 1) # find end of out of schema columns
```

Comment:
You can reuse `manifest_df_header` here rather than computing `manifest_df.columns` again.

Reply:
If I am not mistaken, manifest_df_header contains the old column set (i.e. existing manifest prior update basded on latest schema)?